cocotb/share/lib/verilator/verilator.cpp

```diff
@@ -9,7 +9,11 @@
 #include <memory>
 
 #if VM_TRACE
```

Comment:
Please try to match the new Verilator behavior here, which unconditionally defines VM_TRACE_FST and sets it to 1 or 0. For backwards-compat, introduce a block like this:

```c
#ifndef VM_TRACE_FST
    #define VM_TRACE_FST 0
#endif
```

Reply:
does that do anything useful in that context? 

Shouldn't make a difference even when using #if vs. #ifdef?

Reply:
When you enable FST tracing is `VM_TRACE` defined? I think he is saying you can lift the `if VM_TRACE_FST` out of this `if`.

```c++
#if VM_TRACE_FST
#include <verilated_fst_c.h>
#elif VM_TRACE
#include <verilated_vcd_c.h>
#endif
```

Reply:
Hi, to quote myself:

> The VM_TRACE define is set by verilator and its value cannot be modified.
Cleanest solution seems to be to add another orthogonal define via CFLAGS.

That was the whole point of using an extra define because VM_TRACE always appears.

Reply:
I'm not sure what the problem is with this code. It looks fine to me. I'm partial to merge this anyways unless @imphil explain a bit better. I'll give it 2 days. This is certainly not the last time we will have to deal with Verilator support, it doesn't have to be perfect now, but hopefully will be before 1.5 release at the end of the year.

Reply:
> does that do anything useful in that context?

Yes. The goal is to get exactly the same defines for older and newer Verilator versions.

* For newer Verilator versions, there's unconditionally a define VM_TRACE_FST set to either 0 or 1. 
* For older Verilator versions, you recommend in the docs to set `-DVM_TRACE_FST=1` iff FST tracing is wanted. This means, VM_TRACE_FST is only set to 1 if FST tracing is enabled, and undefined otherwise.

If you add the code snippet I've shown, you get identical behavior.

Now I know that this doesn't make much of a difference given the way the pre-processor treats undefined macros, but I feel it's better to be explicit here.